src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -267,6 +312,14 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
+  const prepareValidityIntervalStart = (validityIntervalStart: any) => {
+    if (!isFeatureSupportedForVersion(deviceVersion, CryptoProviderFeature.VALIDITY_INTERVAL_START)) {
+      const threshold = deviceVersionToStr(TREZOR_VERSIONS[CryptoProviderFeature.VALIDITY_INTERVAL_START])
+      throw Error(`${Errors.OutdatedTrezorValidityIntervalStart} to version ${threshold} or higher`)
+    }
```

Comment:
Do we need a specific error for trezor and ledger, wouldn't a one error suffice? Something like `ValidityIntervalStartNotSupported` ?

And also I would not check whether the interval is undefined in signTxFunction, and instead do something like 
```
const prepareValidityIntervalStart = (validityIntervalStart?: number)
if (validityIntervalStart && !isFeatureSupportedForVersion(deviceVersion, CryptoProviderFeature.VALIDITY_INTERVAL_START))
```
This as well for ttl only with `!ttl` instead of `validityIntervalStart`
What do you think?

Reply:
btw, we could also drop the whole `treshold` functionality and not specify the error to the currently used device, and in that we can use the error as usual 
```
throw Error(Errors.MissingTTLNotSupported)
```
and the message would be something like 
`Optional ttl is supported for ledger version x.x.x and higher and trezor version x.x.x and higher` (phrased better :D)
What do you think?

Reply:
Refactored